### PR TITLE
GHI-364 Refactoring: replace type code with State, replace conditiona…

### PR DESCRIPTION
**Pull request by DmitryBespalov**
…l with polymorphism

progress on #364 to get closer to homogenous statuses across UI and Application service

### CODE REVIEW

File: SafeUIKit/SafeAppUI/AppFlow/MainFlow/TransactionReview/TransactionActionPanelController.swift [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+}
+
+class ReviewCompletedActionPanelController: TransactionActionPanelController {
+
+    override var status: TransactionData.Status { preconditionFailure("Not implemented") }
+
+    override func changeActionPanel(in vc: TransactionReviewViewController) {
+        vc.delegate?.transactionReviewViewControllerDidFinish()
```
Comment (sche) [inline SafeUIKit/SafeAppUI/AppFlow/MainFlow/TransactionReview/TransactionActionPanelController.swift:88]: Is it ok to delegate this delegate call to this controller?

File: SafeUIKit/SafeUIKitDemo/TransactionReviewDemoViewController.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    var controller: TransactionReviewViewController!
+    var walletService = MockWalletApplicationService()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        ApplicationServiceRegistry.put(service: walletService, for: WalletApplicationService.self)
+        walletService.transactionData_output = .default
+        walletService.requestTransactionConfirmation_output = .default
+        walletService.update(account: ethID, newBalance: 90000000000000000)
+        controller = .create()
```
Comment (sche) [inline SafeUIKit/SafeUIKitDemo/TransactionReviewDemoViewController.swift:21]: 👍

**Review by sche — APPROVED** [review]

File: SafeUIKit/SafeAppUI/AppFlow/MainFlow/TransactionReview/TransactionActionPanelController.swift (hunk 1 above)
Comment (DmitryBespalov) [reply SafeUIKit/SafeAppUI/AppFlow/MainFlow/TransactionReview/TransactionActionPanelController.swift:88]: well... for now I think it is ok. not the best thing, agree

File: SafeUIKit/SafeUIKitDemo/TransactionReviewDemoViewController.swift (hunk 2 above)
Comment (DmitryBespalov) [reply SafeUIKit/SafeUIKitDemo/TransactionReviewDemoViewController.swift:21]: 😊

**Pull request merged**